homework01/vector.h

```diff
@@ -1,5 +1,35 @@
 
 class Vector
 {
-	// TODO
+ public:
+  Vector(double x = 0., double y = 0.)
+    {
+      this->x = x;
+      this->y = y;
+    }
+  
+  bool operator==(Vector other)
+  {
+    return (this->GetX() == other.GetX()) && (this->GetY() == other.GetY());
+  }
+  
+  void Add(Vector other)
+  {
+    this->x += other.GetX();
+    this->y += other.GetY();
```

Comment:
Rule of thumb: only one liners in header file - the rest goes to the cc file. The exception would be inlined methods (= performance reason): in this case, you need to keep them in the header file but they should be defined outside the class declaration.